frontend/packages/dev-console/src/utils/__tests__/pipeline-utils.spec.ts

```diff
@@ -56,4 +59,11 @@ describe('pipeline-utils ', () => {
     expect(data.status).not.toBeNull();
     expect(data.status).toBe('Pending');
   });
+
+  it('should return correct params for a pipeline run', () => {
+    const pipelineParams = _.get(mockPipelinesJSON[0], 'spec.params');
+    const params = getPipelineRunParams(pipelineParams);
+    expect(params[0].name).toBe('APP_NAME');
+    expect(params[0].value).toBe('default-app-name');
```

Comment:
Maybe also:
```suggestion
    expect(params[0].value).toBe('default-app-name');
    expect(params[0].default).toBeNotDefined();
```
We are really looking for this to never exist. Another test is also (and probably more-so) accepted.

Reply:
Although we do not need to explicitly test for `default` not being there in `params` since I've added proper types for both `PipelineParams` and `PipelineRunParams`,  I've added an extra test in `pipelines-actions.spec.ts` to explicitly test it.

Reply:
I prefer both tbh. Makes the change harder to accidentally do. But you're right, the type would prevent this issue.